### PR TITLE
CROSSROADS-68-Added RepositoryComponent and ConfigComponent

### PR DESCRIPTION
**Pull request by dvallejo**
This is the first commit of a new module for common functionality.

PR contains:
- *RepositoryComponent*
- *ZookeeperRepositoryComponent* (implementation of *RepositoryComponent* with Curator Framework)
- *ConfigComponent*
- *TypesafeConfigComponent* (implementation of *ConfigComponent* with typesafe config)

Components can be injected as a cake pattern way.

Also, I've added two dummy components (with tests) in order to use it in future tests.

@gserranojc @arinconstrio @JSantosP Could you please review it?

### CODE REVIEW

File: src/main/scala/com/stratio/common/repository/RepositoryComponent.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import scala.util.Try
+
+trait RepositoryComponent[K, V] {
+
+  val repository: Repository
+
+  trait Repository {
+
+    trait RepositoryState
```
Comment (JSantosP) [inline src/main/scala/com/stratio/common/repository/RepositoryComponent.scala]: ```sealed trait```?

File: src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    }
+  }
+
+  object ZookeeperRepository {
+
+    private var curatorClientOpt: Option[CuratorFramework] = None
+
+    def getInstance(config: Config): CuratorFramework = {
+
+      curatorClientOpt.getOrElse {
```
Comment (JSantosP) [inline src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala]: Is it necessary to ```synchronize``` when creating the ```curatorClientOpt``` ?

File: src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import org.json4s.jackson.Serialization.read
+
+import scala.collection.JavaConversions._
+import scala.util.{Failure, Success, Try}
+
+import com.stratio.common.config.ConfigComponent
+import com.stratio.common.repository.RepositoryComponent
+import com.stratio.common.repository.zookeeper.ZookeeperConstants._
+
+trait ZookeeperRepositoryComponent extends RepositoryComponent[String, Array[Byte]] with SLF4JLogging {
```
Comment (JSantosP) [inline src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala]: Maybe not in this branch, but in the following, we could add a ```LoggerComponent``` and express at self-type a dependency on it.

Comment (JSantosP) [inline src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala]: Rest of code just LGTM. Good job :-)

Comment (dvallejo) [inline src/main/scala/com/stratio/common/repository/zookeeper/ZookeeperRepositoryComponent.scala]: Totally agree. Next step will be create a LoggerComponent.
Thanks for the review :)

**Comment by dvallejo** [comment]
@arinconstrio @gserranojc Any more suggestions?

**Comment by arinconstrio** [comment]
LGTM

**Comment by gserranojc** [comment]
+1

**Pull request merged**